widgets/common/src/terraformModal/TerraformModal.tsx

```
@@ -308,6 +308,21 @@ export default function TerraformModal({
             }
         }
 
+        async function createSecretsFromVariables() {
+            const secretActions = new SecretActions(toolbox);
+            const { defaultVisibility } = Consts;
+            const allSecretVariables: Variable[] = [...variables, ...environment].filter(
+                variable => variable.source === 'secret'
+            );
+
+            await allSecretVariables.forEach(async secretVariable => {
+                // add secret if not exist
+                await secretActions.doGet(secretVariable.name).catch(async e => {
+                    await secretActions.doCreate(secretVariable.name, secretVariable.value, defaultVisibility, false);
```

I think performing `doGet` is not necessary - if secret exists `doCreate` could just silently fail. The effect would be the same but we would save on the number of requests.

---

I prefer to keep it this way. I don't want to override the secrets. Now there is no documentation about using creating secrets with existing key. Even if now it returns error I cannot predict that it wouldn't change in the future. https://docs.cloudify.co/api/v3.1/#create-secret